src/hooks/useConnectedReceiver.ts

```diff
@@ -0,0 +1,48 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Parity Bridges UI.
+//
+// Parity Bridges UI is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Parity Bridges UI is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Parity Bridges UI.  If not, see <http://www.gnu.org/licenses/>.
+
+import AccountActions from '../actions/accountActions';
+import { useUpdateAccountContext } from '../contexts/AccountContextProvider';
+import { useSourceTarget } from '../contexts/SourceTargetContextProvider';
+import getReceiverAddress from '../util/getReceiverAddress';
+
+interface Props {
+  setReceiverMessage: (message: string) => void;
+  receiver: string;
+}
+
+export default function useConnectedReceiver(): Function {
+  const { dispatchAccount } = useUpdateAccountContext();
+  const { targetChain } = useSourceTarget();
+
+  const setConnectedReceiver = ({ setReceiverMessage, receiver }: Props) => {
+    setReceiverMessage('');
+    try {
+      const receiverAddress = getReceiverAddress({ chain: targetChain, receiverAddress: receiver });
+      if (receiverAddress !== receiver) {
+        setReceiverMessage(`The format for the account is incorrect, funds will be sent to ${receiverAddress}`);
```

Comment:
We should be more informative with this. I.e. we should parse the SS58 prefixes and display things like: "This address looks like a `Polkadot` address, not `Rialto`'s.".
Also notice there are at least 3 different cases here:
1. Target chain provided - all good (`receiver_address === provided_address`)
2. Source chain provided - derivation (`receiver_addresss === derive(provided_adddress)`)
3. SS58 does not match neither Target nor Source chain - a REALLY FAT warning and we should set `receiver_address = provided_address` (optionally with an option to derive account).

This is actually extremely crucial, so let's have this as a separate issue and make it work extremely well.